tests/test_server.py

```diff
@@ -11,169 +11,194 @@
-    def fin():
-        tmpdir.remove(True)
-
-    tmpdir = path.local(tempfile.mkdtemp())
-    request.addfinalizer(fin)
-    src_setup_py = path.local().join("tests", "centodeps-setup.py")
-    assert src_setup_py.check()
-    projdir = tmpdir.join("centodeps")
-    projdir.mkdir()
-    dst_setup_py = projdir.join("setup.py")
-    src_setup_py.copy(dst_setup_py)
-    assert dst_setup_py.check()
+# A test-distribution to check if
+#    bottle supports uploading 100's of packages,
+#    see: https://github.com/pypiserver/pypiserver/issues/82
+#
+# Has been run once `pip wheel .`, just to generate:
+#    ./wheelhouse/centodeps-0.0.0-cp34-none-win_amd64.whl
+#
+SETUP_PY = """\
+from setuptools import setup
 
-    return projdir
+setup(
+    name="centodeps",
+    install_requires=["a==1.0"] * 200,
+    options={
+        "bdist_wheel": {"universal": True},
+    },
+)
+"""
 
 
 @pytest.fixture(scope="module")
-def package(project, request):
-    with chdir(project.strpath):
-        cmd = "setup.py bdist_wheel"
-        assert _run_python(cmd) == 0
-        pkgs = list(project.join("dist").visit("centodeps*.whl"))
-        assert len(pkgs) == 1
-        pkg = path.local(pkgs[0])
-        assert pkg.check()
+def project(tmp_path_factory):
+    projdir = tmp_path_factory.mktemp("project") / "centodeps"
+    projdir.mkdir(parents=True, exist_ok=True)
+    projdir.joinpath("setup.py").write_text(SETUP_PY)
+    return projdir
+
 
-        return pkg
+@pytest.fixture(scope="session")
+def server_root(tmp_path_factory):
+    return tmp_path_factory.mktemp("root")
 
 
 @pytest.fixture(scope="module")
-def packdir(package):
-    return package.dirpath()
+def wheel_file(project, tmp_path_factory):
+    distdir = tmp_path_factory.mktemp("dist")
+    assert run_setup_py(project, f"bdist_wheel -d {distdir}") == 0
+    return list(distdir.glob("centodeps*.whl"))[0]
+
 
+@pytest.fixture()
+def hosted_wheel_file(wheel_file, server_root):
+    dst = server_root / wheel_file.name
+    shutil.copy(wheel_file, dst)
+    yield dst
+    if dst.is_file():
+        dst.unlink()
 
-open_port = 8081
+
+def clear_directory(root: Path):
+    for path in root.iterdir():
+        if path.is_file():
+            path.unlink()
 
 
 @pytest.fixture(scope="module")
-def open_server(packdir, request):
-    srv = _run_server(packdir, open_port, authed=False)
-    fin = functools.partial(_kill_server, srv)
-    request.addfinalizer(fin)
+def _open_server(server_root):
+    with run_server(server_root, authed=False) as srv:
+        yield srv
 
-    return srv
 
+@pytest.fixture
+def open_server(_open_server: Srv):
+    yield _open_server
+    clear_directory(_open_server.root)
+
+
+@pytest.fixture(scope="module")
+def _authed_server(server_root):
+    with run_server(server_root, authed=True) as srv:
+        yield srv
 
-protected_port = 8082
+
+@pytest.fixture
+def authed_server(_authed_server):
+    yield _authed_server
+    clear_directory(_authed_server.root)
```

Comment:
Nice use of the module and function-scoped fixtures to keep a long-lived server while ensuring a fresh state for each test.